src/listconf.c

```diff
@@ -397,11 +397,6 @@ void listconf_parse_early(void)
 			exit(EXIT_FAILURE);
 		}
 	}
-	if (!strcasecmp(options.listconf, "hidden-options"))
-	{
-		opt_print_hidden_usage();
-		exit(EXIT_SUCCESS);
-	}
```

Comment:
I think for compatibility with existing scripts/front-ends we need to support the deprecated `--list=hidden-options` but it should now be the same as just saying `--help`.

Reply:
So I've been looking into that and would like to ask for some feedback what you think would be the best way to go on about that?

What comes to mind is moving the PRINT_HELP macros to a help.h or print_help.h and create a print_help(); function that I import in options.c.

Reply:
I'd keep them in `options.c` (so eg. `opt_help()`) but yes, an exported function that we can call from here.